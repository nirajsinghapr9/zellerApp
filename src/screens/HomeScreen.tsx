import React, {useState} from 'react';
import {
  View,
  Text,
  FlatList,
  TouchableOpacity,
  StyleSheet,
  ActivityIndicator,
} from 'react-native';
import {useQuery} from '@apollo/client';
import {LIST_ZELLER_CUSTOMERS} from '../graphql/queries';
import {
  ZellerCustomer,
  ListZellerCustomersData,
  ListZellerCustomersVars,
} from '../types/graphqlTypes';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RadioButton} from 'react-native-paper';
import Avatar from './components/Avatar';
import {capitalizeFirstChar} from '../util/util';

type RootStackParamList = {
  Home: undefined;
  Second: undefined;
};

type Props = NativeStackScreenProps<RootStackParamList, 'Home'>;

const USER_TYPES = ['Admin', 'Manager'];

export default function HomeScreen({navigation}: Props) {
  const [selectedRole, setSelectedRole] = useState<string>('Admin');

  const {data, loading, error, refetch} = useQuery<
    ListZellerCustomersData,
    ListZellerCustomersVars
  >(LIST_ZELLER_CUSTOMERS, {
    variables: selectedRole
      ? {filter: {role: {eq: selectedRole.toUpperCase()}}}
      : {},
  });

  function renderItem({item}: {item: ZellerCustomer}) {
    return (
      <TouchableOpacity
        style={styles.item}
        onPress={() => navigation.navigate('Second')}>
        <Avatar name={item.name} />
        <View style={styles.itemTextContainer}>
          <Text style={styles.name}>{item.name}</Text>
          <Text style={styles.role}>
            {capitalizeFirstChar(item?.role || '')}
          </Text>
        </View>
      </TouchableOpacity>
    );
  }

  return (
    <View style={styles.container}>
      <View style={styles.filterContainer}>
        <Text style={styles.headerTitle}>User Types</Text>
        {USER_TYPES.map(role => (
          <TouchableOpacity
            key={role}
            style={[
              styles.radioButtonContainer,
              selectedRole === role && styles.radioButtonSelected,
            ]}
            onPress={() => setSelectedRole(role)}
            testID={`radio-${role}`}
            activeOpacity={0.7}>
            <RadioButton
              value={role}
              status={selectedRole === role ? 'checked' : 'unchecked'}
              onPress={() => setSelectedRole(role)}
            />
            <Text style={styles.name}>{role}</Text>
          </TouchableOpacity>
        ))}
      </View>

      <View style={styles.divider} />

      {loading && (
        <ActivityIndicator
          size="large"
          color="#0000ff"
          testID="loading-indicator"
        />
      )}
      {error && <Text style={styles.error}>Error loading users.</Text>}

      <FlatList
        onRefresh={refetch}
        refreshing={loading}
        data={data?.listZellerCustomers.items ?? []}
        keyExtractor={item => item.id}
        renderItem={renderItem}
        testID="user-list"
        contentContainerStyle={styles.listContainer}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {flex: 1, padding: 16, backgroundColor: '#fff'},
  filterContainer: {
    marginBottom: 16,
  },

  item: {
    padding: 12,
    marginVertical: 6,
    borderRadius: 8,
    flexDirection: 'row',
  },
  name: {fontSize: 16, color: 'black'},
  role: {fontSize: 12, color: '#999'},
  error: {color: 'red', marginVertical: 12},
  listContainer: {paddingBottom: 80},
  navButton: {
    marginTop: 20,
    backgroundColor: '#007AFF',
    padding: 12,
    borderRadius: 8,
    alignItems: 'center',
  },

  radioButtonContainer: {flexDirection: 'row', alignItems: 'center'},
  radioButtonSelected: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#e6f7ff',
    borderRadius: 10,
  },
  headerTitle: {
    fontSize: 20,
    color: 'black',
    fontWeight: '500',
    marginBottom: 10,
  },
  divider: {
    height: 1,
    backgroundColor: '#e6e6e6',
    marginVertical: 20,
  },
  itemTextContainer: {
    flexDirection: 'column',
    marginLeft: 12,
  },
});
