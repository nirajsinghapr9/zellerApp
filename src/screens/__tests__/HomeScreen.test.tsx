import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react-native';
import HomeScreen from '../HomeScreen';

// Mock React Navigation's useNavigation
jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

const mockRoute = {
  key: 'HomeScreen',
  name: 'Home',
  params: undefined,
};

const mockUseQuery = jest.fn();
jest.mock('@apollo/client', () => {
  return {
    gql: (strings: TemplateStringsArray) => strings[0],
    useQuery: (query: any, options: any) => mockUseQuery(query, options),
  };
});

// Dummy data for test
const mockData = {
  listZellerCustomers: {
    items: [
      {id: '1', name: 'Alice', role: 'ADMIN'},
      {id: '2', name: 'Bob', role: 'MANAGER'},
    ],
  },
};

describe('HomeScreen', () => {
  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it('shows loading indicator while loading', () => {
    mockUseQuery.mockReturnValue({
      data: undefined,
      loading: true,
      error: undefined,
      refetch: jest.fn(),
    });

    const {toJSON} = render(
      <HomeScreen
        navigation={{navigate: jest.fn()} as any}
        route={mockRoute as any}
      />,
    );
    expect(screen.getByTestId('loading-indicator')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('shows error message if error occurs', () => {
    mockUseQuery.mockReturnValue({
      data: undefined,
      loading: false,
      error: new Error('Failed to fetch'),
      refetch: jest.fn(),
    });

    const {getByText} = render(
      <HomeScreen
        navigation={{navigate: jest.fn()} as any}
        route={mockRoute as any}
      />,
    );
    expect(getByText('Error loading users.')).toBeTruthy();
  });

  it('renders list of users', () => {
    mockUseQuery.mockReturnValue({
      data: mockData,
      loading: false,
      error: undefined,
      refetch: jest.fn(),
    });

    const {toJSON} = render(
      <HomeScreen
        navigation={{navigate: jest.fn()} as any}
        route={mockRoute as any}
      />,
    );

    expect(screen.getByText('Alice')).toBeTruthy();
    expect(screen.getByText('Bob')).toBeTruthy();
    expect(screen.getAllByText(/Admin|Manager/i).length).toBeGreaterThanOrEqual(
      1,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('changes selected role when clicking radio buttons', () => {
    mockUseQuery.mockReturnValue({
      data: mockData,
      loading: false,
      error: undefined,
      refetch: jest.fn(),
    });

    const {getByTestId, toJSON} = render(
      <HomeScreen
        navigation={{navigate: jest.fn()} as any}
        route={mockRoute as any}
      />,
    );
    expect(screen.getByTestId('radio-Admin')).toBeDefined();
    expect(screen.getByTestId('radio-Manager')).toBeDefined();

    const managerRadio = getByTestId('radio-Manager');

    fireEvent.press(managerRadio);
    expect(toJSON()).toMatchSnapshot();
  });
});
