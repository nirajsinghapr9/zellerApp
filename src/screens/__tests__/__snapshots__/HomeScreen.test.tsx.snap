// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen changes selected role when clicking radio buttons 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 10,
        }
      }
    >
      User Types
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="radio-Admin"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 0,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Admin
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e6f7ff",
          "borderRadius": 10,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="radio-Manager"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": true,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Manager
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e6e6e6",
        "height": 1,
        "marginVertical": 20,
      }
    }
  />
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 80,
      }
    }
    data={
      [
        {
          "id": "1",
          "name": "Alice",
          "role": "ADMIN",
        },
        {
          "id": "2",
          "name": "Bob",
          "role": "MANAGER",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[MockFunction]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="user-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 8,
              "flexDirection": "row",
              "marginVertical": 6,
              "opacity": 1,
              "padding": 12,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 4,
                  "justifyContent": "center",
                },
                {
                  "backgroundColor": "#e6f7ff",
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0099e6",
                    "fontWeight": "bold",
                  },
                  {
                    "fontSize": 20,
                  },
                ]
              }
            >
              A
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "column",
                "marginLeft": 12,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontSize": 16,
                }
              }
            >
              Alice
            </Text>
            <Text
              style={
                {
                  "color": "#999",
                  "fontSize": 12,
                }
              }
            >
              Admin
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 8,
              "flexDirection": "row",
              "marginVertical": 6,
              "opacity": 1,
              "padding": 12,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 4,
                  "justifyContent": "center",
                },
                {
                  "backgroundColor": "#e6f7ff",
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0099e6",
                    "fontWeight": "bold",
                  },
                  {
                    "fontSize": 20,
                  },
                ]
              }
            >
              B
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "column",
                "marginLeft": 12,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontSize": 16,
                }
              }
            >
              Bob
            </Text>
            <Text
              style={
                {
                  "color": "#999",
                  "fontSize": 12,
                }
              }
            >
              Manager
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`HomeScreen renders list of users 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 10,
        }
      }
    >
      User Types
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e6f7ff",
          "borderRadius": 10,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="radio-Admin"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": true,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Admin
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="radio-Manager"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 0,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Manager
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e6e6e6",
        "height": 1,
        "marginVertical": 20,
      }
    }
  />
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 80,
      }
    }
    data={
      [
        {
          "id": "1",
          "name": "Alice",
          "role": "ADMIN",
        },
        {
          "id": "2",
          "name": "Bob",
          "role": "MANAGER",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[MockFunction]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="user-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 8,
              "flexDirection": "row",
              "marginVertical": 6,
              "opacity": 1,
              "padding": 12,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 4,
                  "justifyContent": "center",
                },
                {
                  "backgroundColor": "#e6f7ff",
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0099e6",
                    "fontWeight": "bold",
                  },
                  {
                    "fontSize": 20,
                  },
                ]
              }
            >
              A
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "column",
                "marginLeft": 12,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontSize": 16,
                }
              }
            >
              Alice
            </Text>
            <Text
              style={
                {
                  "color": "#999",
                  "fontSize": 12,
                }
              }
            >
              Admin
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 8,
              "flexDirection": "row",
              "marginVertical": 6,
              "opacity": 1,
              "padding": 12,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 4,
                  "justifyContent": "center",
                },
                {
                  "backgroundColor": "#e6f7ff",
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0099e6",
                    "fontWeight": "bold",
                  },
                  {
                    "fontSize": 20,
                  },
                ]
              }
            >
              B
            </Text>
          </View>
          <View
            style={
              {
                "flexDirection": "column",
                "marginLeft": 12,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontSize": 16,
                }
              }
            >
              Bob
            </Text>
            <Text
              style={
                {
                  "color": "#999",
                  "fontSize": 12,
                }
              }
            >
              Manager
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`HomeScreen shows loading indicator while loading 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 10,
        }
      }
    >
      User Types
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e6f7ff",
          "borderRadius": 10,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="radio-Admin"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": true,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Admin
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID="radio-Manager"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
              "padding": 6,
            },
          ]
        }
      >
        <View
          style={
            {
              "opacity": 0,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 16,
          }
        }
      >
        Manager
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e6e6e6",
        "height": 1,
        "marginVertical": 20,
      }
    }
  />
  <ActivityIndicator
    color="#0000ff"
    size="large"
    testID="loading-indicator"
  />
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 80,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[MockFunction]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={true}
      />
    }
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="user-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
